lib/ansible/modules/notification/telegram.py

```diff
@@ -69,25 +75,42 @@
   returned: success
   type: string
   sample: "Ansible task finished"
+telegram_error:
+  description: Error message gotten from Telegram API
+  returned: failure
+  type: string
+  sample: "Bad Request: message text is empty"
 """
 
-import urllib
+try:
+    from urllib import quote
+except ImportError:
+    from urllib.parse import quote
+
+import json
+
 
 def main():
 
     module = AnsibleModule(
-        argument_spec = dict(
-            token = dict(type='str',required=True,no_log=True),
-            chat_id = dict(type='str',required=True,no_log=True),
-            msg = dict(type='str',required=True)),
+        argument_spec=dict(
+            token=dict(type='str', required=True, no_log=True),
+            chat_id=dict(type='str', required=True, no_log=True),
+            msg_format=dict(type='str', required=False, default='plain',
+                            choices=['plain', 'markdown', 'html']),
+            msg=dict(type='str', required=True)),
         supports_check_mode=True
     )
 
-    token = urllib.quote(module.params.get('token'))
-    chat_id = urllib.quote(module.params.get('chat_id'))
-    msg = urllib.quote(module.params.get('msg'))
+    token = quote(module.params.get('token'))
+    chat_id = quote(module.params.get('chat_id'))
+    msg_format = quote(module.params.get('msg_format'))
+    msg = quote(module.params.get('msg'))
 
-    url = 'https://api.telegram.org/' + token + '/sendMessage?text=' + msg + '&chat_id=' + chat_id
```

Comment:
Please fix url to` 'https://api.telegram.org/bot' + token + '/sendMessage?text='`
and examples descriptions to `token: '9999999:XXXXXXXXXXXXXXXXXXXXXXX'`
Because BotFather return token without "bot" string